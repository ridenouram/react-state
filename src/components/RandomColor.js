import React, { PureComponent } from 'react';
import randomColor from 'randomcolor';

export default class RandomColor extends PureComponent {
  state = {
    color: ''
  }

  //change color function
  randomize = () => {
    const color = randomColor();
    this.setState({ color });
  }

  //life cycle
  componentDidMount() {
    setInterval(this.randomize, 1000);
  }

  //renders div
  render() {
    const { color } = this.state;

    return (
      <div style={{ background: color, width: '100px', height: '100px' }} />
    );
  }
}
