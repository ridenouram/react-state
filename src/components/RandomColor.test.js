import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

describe('RandomColor component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<RandomColor />);
    expect(wrapper).toMatchSnapshot();
  });
});
