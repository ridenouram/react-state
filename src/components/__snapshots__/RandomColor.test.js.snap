// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomColor component matches snapshot 1`] = `
<div
  style={
    Object {
      "background": "",
      "height": "100px",
      "width": "100px",
    }
  }
/>
`;
